internal/validation/validation_test.go

```diff
@@ -75,4 +104,7 @@ func sortLocations(errs []*errors.QueryError) {
 		locs := err.Locations
 		sort.Slice(locs, func(i, j int) bool { return locs[i].Before(locs[j]) })
 	}
+
+	// FIXME: Sorting errors seems necessary, as mutations being validated before queries, but order reversed in expectations?
```

Comment:
Then we can't really avoid it, I guess

Reply:
This was a note to come back to / discuss once better understood. Could be the order doesn't really matter here, and sorting before comparison is right, or could be changing the order things are evaluated for validation is needed. Will try to dig in here

Reply:
Resolving this was a case of changing _when_ the directives were validated for the operations from _before_ validating the variables, to _after_. Given it was a simple enough change, I've opted to do that, and remove the sorting of the errors (sorting of locations is left as it was)